### PR TITLE
added tests to inforgraphic api

### PR DESCRIPTION
**Pull request by DavidShemian**

### CODE REVIEW

File: tests/test_infographic_api.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
+import pytest
+import six
+from six.moves import http_client
```
Comment (idosavion) [inline tests/test_infographic_api.py:3]: You probably should not use six

File: tests/test_infographic_api.py [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+@pytest.fixture
+def app():
+    return flask_app.test_client()
+
+
+if six.PY2:
+    def _text_data(rv): return rv.data
+else:
+    def _text_data(rv): return rv.data.decode("utf-8")
```
Comment (idosavion) [inline tests/test_infographic_api.py:16]: we're in python3

File: tests/test_infographic_api.py [hunk 3]
```diff
@@ -0,0 +1,304 @@
+import pytest
```
Comment (idosavion) [inline tests/test_infographic_api.py:1]: Can you please use black before pushing this file?

File: tests/test_infographic_api.py [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+
+    @pytest.fixture
+    def widgets(self, infographic_from_existing_new_flash):
+        return infographic_from_existing_new_flash['widgets']
+
+    def test_no_news_flash_id(self, app):
+        '''Should success and be empty when no flash id is sent'''
+        rv = app.get('/api/infographics-data')
+        assert rv.status_code == http_client.OK
+        assert not _text_data(rv)
```
Comment (idosavion) [inline tests/test_infographic_api.py:37]: Those asserts will be might be optimized away, you should use `self.assert*` methods.

File: tests/test_infographic_api.py [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+
+    def test_location_info(self, infographic_from_existing_new_flash):
+        assert infographic_from_existing_new_flash['meta']['location_info'] == {
+            "resolution": "כביש בינעירוני",
+            "road1": 90.0,
+            "road_segment_name": "כניסה למצפה שלם - צומת שדי תרומות"
+        }
+
+    def test_infographic_with_existing_news_flash(self, widgets):
```
Comment (idosavion) [inline tests/test_infographic_api.py:52]: IMO the widget should be a constant, 
All those tests are going to break the moment the value return by the website will be different,
while the functionality under tests will stay the same.

File: tests/test_infographic_api.py [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+        self.__accident_count_by_accident_year_test(widgets)
+        self.__injured_count_by_accident_year_test(widgets)
+        self.__accident_count_by_day_night_test(widgets)
+        self.__accidents_count_by_hour_test(widgets)
+        self.__accident_count_by_road_light_test(widgets)
+        self.__top_road_segments_accidents_per_km_test(widgets)
+        self.__injured_count_per_age_group_test(widgets)
+        self.__injured_vision_zero_test(widgets)
+
+    def __accident_count_by_severity_test(self, widgets):
```
Comment (idosavion) [inline tests/test_infographic_api.py:69]: why the need for double underscore?

File: tests/test_infographic_api.py (hunk 1 above)
Comment (DavidShemian) [reply tests/test_infographic_api.py:3]: Done

File: tests/test_infographic_api.py (hunk 2 above)
Comment (DavidShemian) [reply tests/test_infographic_api.py:16]: Done

File: tests/test_infographic_api.py (hunk 3 above)
Comment (DavidShemian) [reply tests/test_infographic_api.py:1]: Done

File: tests/test_infographic_api.py (hunk 4 above)
Comment (DavidShemian) [reply tests/test_infographic_api.py:33]: No need, as we spoke about

File: tests/test_infographic_api.py (hunk 5 above)
Comment (DavidShemian) [reply tests/test_infographic_api.py:48]: We talked about this

File: tests/test_infographic_api.py (hunk 6 above)
Comment (DavidShemian) [reply tests/test_infographic_api.py:70]: Change to single underscore

**Comment by idosavion** [comment]
> @DavidShemian great work!
> Regarding the tests that failed:
> 
> 1. **test_no_news_flash_id** : Nice catch on this one. Right now we just return the mock data even if there is no news_flash_id. What should we return in this case of accessing /api/infographics-data with no news_flash_id? Perhaps is should be status 400 - Bad Request?
>    @ziv17 what do you think?
> 2. **test_bad_news_flash_id**: Same here - great catch. If news flash id does not exist, I assume we'll want to return 400 as well. @ziv17 any thoughts? Also, once we insert @ziv17 's PR, I suggest we'll stop using the mock data at all. @ziv17 what do you think?
> 3. **test_location_info**: So this one test is failing, since the news flash is not in the db - so currently you get the mock data. How about inserting it and then testing? Hence you need to add the data of this news flash to the db (to the NewsFlash table), and then you'll be able to test the location info and this test should not fail.
> 4. **test_infographic_api**: This test fails due to the fact that not all CBS data is indeed loaded to the DB during these tests, only partial data of 2014. I suggest you'll match the number to the one of the partial data for now (86 and change to >=).
> 
> @idosavion @elazarg Any other thoughts?

I already talked about my comments with David, I don't like the fact that the test depends on the DB state, but I can't think of a better solution at the moment :(

**Comment by ziv17** [comment]
>     1. ziv17

**Comment by ziv17** [comment]
> 
> 
> @DavidShemian great work!
> Regarding the tests that failed:
> 
>     1. **test_no_news_flash_id** : Nice catch on this one. Right now we just return the mock data even if there is no news_flash_id. What should we return in this case of accessing /api/infographics-data with no news_flash_id? Perhaps is should be status 400 - Bad Request?
>        @ziv17 what do you think?
> 
>     2. **test_bad_news_flash_id**: Same here - great catch. If news flash id does not exist, I assume we'll want to return 400 as well. @ziv17 any thoughts? Also, once we insert @ziv17 's PR, I suggest we'll stop using the mock data at all. @ziv17 what do you think?
> 
>     3. **test_location_info**: So this one test is failing, since the news flash is not in the db - so currently you get the mock data. How about inserting it and then testing? Hence you need to add the data of this news flash to the db (to the NewsFlash table), and then you'll be able to test the location info and this test should not fail.
> 
>     4. **test_infographic_api**: This test fails due to the fact that not all CBS data is indeed loaded to the DB during these tests, only partial data of 2014. I suggest you'll match the number to the one of the partial data for now (86 and change to >=).
> 
> 
> @idosavion @elazarg Any other thoughts?

regarding 1:  Agree.  400 bad request is appropriate here.
Regarding 2: Maybe 404, not found? As you suggested, I think infographics data should return an empty response, and write an error to our log.

**Comment by atalyaalon** [comment]
@ziv17 Thanks!
@DavidShemian - please perform relevant fixes and update this PR.
Thanks!

**Comment by atalyaalon** [comment]
@DavidShemian any update on this one?

**Pull request merged**